### ISSUE TITLE
6157 - staging screen rendering bug

### ISSUE THREAD

**Issue by DanVanAtta** [triplea-game/triplea#2220]
After clicking 'leave game', the staging screen is rendered as this:

![staging_screen_rendering_bug](https://user-images.githubusercontent.com/12397753/29330901-91613a40-81af-11e7-8695-9f13a62cb430.png)

**Comment by ron-murhammer** [comment]
@DanVanAtta Funky. I've never clicked 'Leave Game'. I always just click the X in the top corner which works fine. It appears 'Leave Game' does something different...

**Comment by ssoloff** [comment]
@DanVanAtta @ron-murhammer When you guys return to the staging screen, are the Resource Modifiers controls already expanded?  Whenever I return, even if I set them to something other than the default values, the controls are always collapsed.

I'm asking because I don't observe this problem, and I'm wondering if it has something to do with these controls being initially expanded vs. collapsed.

**Comment by ron-murhammer** [comment]
@ssoloff I only see the issue if I expanded resource modifiers before launching the game. It appears to try and re-render the screen after leaving the game without cleaning it or something.

**Comment by ssoloff** [comment]
@ron-murhammer Yeah, I tried that before, and I still couldn't reproduce it.

I just tried again, and this time I noticed something interesting.  Immediately upon returning to the staging screen, I see the Resource Modifiers controls expanded for a fraction of a second, and then they are immediately collapsed.  I'm running this on a VM, so it's possible there's a race condition, and my VM is slowing things down enough for the repaint to happen correctly.

Yup, I think that's it.  I went over to my physical box, and I'm able to reproduce the problem consistently there.  So, I wouldn't be surprised if there's a `SwingUtilities.invokeLater()` call someplace that is responsible for this issue.
